ERA5/construct_era5_daily_cancities_revised.py

```diff
@@ -0,0 +1,444 @@
+        for variable_folder in Path(NAMpath.as_posix().format(time=freq)).iterdir():
+            variable = variable_folder.name
+            fpath = Path(NAMpath.as_posix().format(time=freq)).joinpath(variable)
+            if fpath.exists():
+                logging.info(f"Found {freq} folders for {variable}.")
+                for y in range(4):
+                    files.extend(list(fpath.glob((glob_files.format(time=freq, variable=variable, y=y)))))
+
+    raw_hrly_nam = xr.open_mfdataset(
+        gather["1hr"],
+        chunks={"time": 2928, "latitude": 25, "longitude": 50},
+        engine="zarr"
+    )
+    raw_dly_nam = xr.open_mfdataset(
+        gather["day"],
+        chunks={"time": 2928, "latitude": 25, "longitude": 50},
+        engine="zarr"
+    )
+
+    location = xr.DataArray(
+        ["Halifax", "Montréal", "Iqaluit", "Saskatoon", "Victoria"],
+        dims=("location",),
+        name="location",
+        attrs={"long_name": "City"},
+    )
+    lon = xr.DataArray(
+        [-63.5, -73.5, -68.5, -106.75, -123.25],
+        dims=("location",),
+        coords={"location": location},
+        attrs={
+            "standard_name": "longitude",
+            "units": "degree_east",
+            "long_name": "longitude",
+        },
+    )
+    lat = xr.DataArray(
+        [44.5, 45.5, 63.75, 52.0, 48.5],
+        dims=("location",),
+        coords={"location": location},
+        attrs={
+            "standard_name": "latitude",
+            "units": "degree_north",
+            "long_name": "latitude",
+        },
+    )
+
+    hrly = raw_hrly_nam.sel(lon=lon, lat=lat, method="nearest")
+    dly = raw_dly_nam.sel(lon=lon, lat=lat, method="nearest")
+
+    dly.lon.attrs.update(lon.attrs)
+    dly.lat.attrs.update(lat.attrs)
+
+    # Variables computation
+    # Loosely regrouped by thematics
+
+    if "tas" not in dly.data_vars:
+        tas = hrly.tas.resample(time="D").mean()
+    else:
+        tas = dly.tas
+
+    if "tasmin" not in dly.data_vars:
+        tasmin = hrly.tas.resample(time="D").min()
+    else:
+        tasmin = dly.tasmin
+
+    if "tasmax" not in dly.data_vars:
+        tasmax = hrly.tas.resample(time="D").max()
+    else:
+        tasmax = dly.tasmax
+
+    tas.attrs.update(
+        standard_name="air_temperature",
+        long_name="Mean daily surface temperature",
+        units="K",
+        cell_methods="time: mean within days",
+    )
+    tasmin.attrs.update(
+        standard_name="air_temperature",
+        long_name="Minimum daily surface temperature",
+        units="K",
+        cell_methods="time: minimum within days",
+    )
+    tasmax.attrs.update(
+        standard_name="air_temperature",
+        long_name="Mean daily surface temperature",
+        units="K",
+        cell_methods="time: maximum within days",
+    )
+
+    if "pr" not in dly.data_vars:
+        # Total precip flux in kg m-2 s-1
+        pr = hrly.pr.resample(time="D").sum()
```

Comment:
```suggestion
        pr = hrly.pr.resample(time="D").mean()
```
If it is a flux, then the resampling should be a mean, no?

Reply:
Good catch. It shouldn't be an issue for the existing data as `dly` exists, but yes, absolutely must be fixed!